sklearn/preprocessing/_data.py

```diff
@@ -3191,7 +3212,10 @@ def _neg_log_likelihood(lmbda):
         # get rid of them
         x = x[~np.isnan(x)]
         # choosing bracket -2, 2 like for boxcox
-        return optimize.brent(_neg_log_likelihood, brack=(-2, 2))
+        min_lmbda, min_ll, _, _ = optimize.brent(
+            _neg_log_likelihood, brack=(-2, 2), full_output=True
+        )
+        return min_lmbda, min_ll
```

Comment:
this can probably be reverted?

Reply:
How would you tell if the optimization has failed otherwise?

Maybe we could use it to write a unit test for that `_yeo_johnson_optimize` to check that is `-np.inf` for near constant data but finite for non-constant data.

Reply:
oh my bad , I thought `brent()` was returning min_lmbda and min_ll already